core/domain/suggestion_registry.py

```diff
@@ -0,0 +1,332 @@
+        self.target_type = suggestion_models.TARGET_TYPE_EXPLORATION
+        self.target_id = target_id
+        self.target_version_at_submission = target_version_at_submission
+        self.status = status
+        self.author_id = author_id
+        self.assigned_reviewer_id = assigned_reviewer_id
+        self.final_reviewer_id = final_reviewer_id
+        self.change_cmd = change_cmd
+        self.score_category = score_category
+
+    def validate(self):
+        """Validates a suggestion object of type
+        SUGGESTION_TYPE_EDIT_STATE_CONTENT.
+
+        Raises:
+            ValidationError: One or more attributes of the
+                SuggestionEditStateContent object are invalid.
+        """
+        super(SuggestionEditStateContent, self).validate()
+
+        if self.get_score_type() != suggestion_models.SCORE_TYPE_CONTENT:
+            raise utils.ValidationError(
+                'Expected the first part of score_category to be %s '
+                ', received %s' % (
+                    suggestion_models.SCORE_TYPE_CONTENT,
+                    self.get_score_type()))
+
+        if self.get_score_sub_type() not in constants.ALL_CATEGORIES:
+            raise utils.ValidationError(
+                'Expected the second part of score_category to be a valid'
+                ' category, received %s' % self.get_score_sub_type())
+
+        if 'cmd' not in self.change_cmd:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain a cmd key')
+
+        if self.change_cmd['cmd'] != exp_domain.CMD_EDIT_STATE_PROPERTY:
+            raise utils.ValidationError(
+                'Expected cmd to be %s, received %s' % (
+                    exp_domain.CMD_EDIT_STATE_PROPERTY, self.change_cmd['cmd']))
+
+        if 'property_name' not in self.change_cmd:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain a property_name key')
+
+        if (
+                self.change_cmd['property_name'] !=
+                exp_domain.STATE_PROPERTY_CONTENT):
+            raise utils.ValidationError(
+                'Expected property_name to be %s, received %s' % (
+                    exp_domain.STATE_PROPERTY_CONTENT,
+                    self.change_cmd['property_name']))
+
+        if 'state_name' not in self.change_cmd:
+            raise utils.ValidationError(
+                'Expected change_cmd to contain a state_name key')
+
+        states = exp_services.get_exploration_by_id(self.target_id).states
+        if self.change_cmd['state_name'] not in states:
+            raise utils.ValidationError(
+                'Expected %s to be a valid state name' %
+                self.change_cmd['state_name'])
+
+    def update_change_cmd_before_accept(self):
+        """Before accepting the suggestion, modifications need to be done to the
+        change_cmd.
+        """
+        exploration = exp_services.get_exploration_by_id(self.target_id)
```

Comment:
Better to pass the old_content in as an argument, if possible.

Also don't modify the domain object, do something like get_change_list_for_accepting_suggestion() or similar and let the *_services.py file handle all the operations like calling update_exploration() etc.

Reply:
Actually I think passing it separately won't work. For every suggestion type the old content is obtained differently. So I think obtaining it within this function is a better idea for generalizing. What's your opinion on this?

Also I did make a get_change_list_for_accepting_suggestion function which I call just before accepting. Another thing I am doubtful about is letting the services file handle update_exploration. This again relates to generalisation. It would be better to have it in the accept function in the domain object itself as it is type specific.